habitat-lab/habitat/core/embodied_task.py

```diff
@@ -295,45 +295,43 @@ def reset(self, episode: Episode):
 
     def _step_single_action(
         self,
-        observations: Any,
         action_name: Any,
         action: Dict[str, Any],
         episode: Episode,
-        is_last_action=True,
     ):
         if isinstance(action_name, (int, np.integer)):
             action_name = self.get_action_name(action_name)
         assert (
             action_name in self.actions
         ), f"Can't find '{action_name}' action in {self.actions.keys()}."
         task_action = self.actions[action_name]
-        observations.update(
-            task_action.step(
-                **action["action_args"],
-                task=self,
-                is_last_action=is_last_action,
-            )
+        return task_action.step(
+            **action["action_args"],
+            task=self,
         )
 
     def step(self, action: Dict[str, Any], episode: Episode):
         action_name = action["action"]
         if "action_args" not in action or action["action_args"] is None:
             action["action_args"] = {}
-        observations: Any = {}
+        observations: Any = None
         if isinstance(action_name, tuple):  # there are multiple actions
-            for i, a_name in enumerate(action_name):
-                self._step_single_action(
-                    observations,
+            for a_name in action_name:
+                observations = self._step_single_action(
                     a_name,
                     action,
                     episode,
-                    i == len(action_name) - 1,
                 )
         else:
-            self._step_single_action(
-                observations, action_name, action, episode
+            observations = self._step_single_action(
+                action_name, action, episode
             )
 
+        self._sim.step_physics(1.0 / 60.0)  # type:ignore
```

Comment:
What is this call? Why is this using a fixed 60Hz? This didn't seem to be here before, why add it?

Reply:
The simulator method always steps physics with a dt of 1/60 (see [here](https://github.com/facebookresearch/habitat-sim/blob/main/src_python/habitat_sim/simulator.py#L503))
Before, each action was calling sim.step which is INCREDIBLY INEFFICIENT WHEN THERE ARE MORE THAN ONE ACTION, so instead, the actions now just perform the action and so not step the simulator. The stepping of the simulator happens in `EmbodiedTask`. Since there is no action to perform and the simulator just the physics needs to be stepped, we do not call sim.step (since this method takes as input an action that does not exist) but instead, we manually step the physics with sim.step_physics.

Reply:
Agreed this step size should be configured somewhere rather than locked at 60Hz.
True this is the default, but better that it is configured than a magic number.